prime-router/src/main/kotlin/Element.kt

```diff
@@ -381,6 +386,34 @@ data class Element(
         return truncateIfNeeded(formattedValue)
     }
 
+    // this method takes a date value as a string and returns a
+    // TemporalAccessor based on the variable date time pattern
+    fun parseDate(dateValue: String): TemporalAccessor {
+        return DateTimeFormatter.ofPattern(variableDateTimePattern)
+            .parseBest(dateValue, OffsetDateTime::from, LocalDateTime::from, Instant::from, LocalDate::from)
+    }
+
+    // given a temporal accessor this will check the type that it needs to return
+    // and then output based on the format. you can extend this to accept a third
+    // variable which would be the element's output format, and do an extra branch
+    // based on that
+    fun getDate(temporalAccessor: TemporalAccessor, outputFormat: String): String {
+        val outputFormatter = DateTimeFormatter.ofPattern(outputFormat)
+        val formattedDate = when {
+            // if the date parsed out as LocalDate
+            temporalAccessor is LocalDate ->
+                LocalDate.from(temporalAccessor)
+                    .atStartOfDay()
+                    .format(outputFormatter)
+            temporalAccessor is LocalDateTime ->
+                LocalDateTime.from(temporalAccessor)
+                    .format(outputFormatter)
+            else -> error("Unsupported format!")
```

Comment:
Ott, you should probably add logic to also process the `OffsetDateTime` and `Instant` since we can parse those out too.

Reply:
Done.